### PR TITLE
im just a chill guy

### PR DESCRIPTION
**Pull request by R0gue-one**
Added resell feature and some minor fixes that i dont remember

solves issue: #13 
possibly solves: #16

### CODE REVIEW

File: backend/app/routes/admin_routes.py [hunk 1]
```diff
@@ -55,12 +55,12 @@ def expire_credit(credit_id):
...
     user = User.query.filter_by(username=current_user.get('username')).first()
     credit = Credit.query.get(credit_id)
-    pc = PurchasedCredit.query.get(credit_id)
+    pc = PurchasedCredit.query.filter_by(credit_id=credit.id).first()
 
     if not credit:
         return jsonify({"message": "Credit not found"}), 404
     if not pc:
-        return jsonify({"message": "Credit can't be expired as it has not been sold yet"}), 400
+        return jsonify({"message": f"Credit can't be expired as it has not been sold yet, credit with B_ID {credit_id} is not found"}), 400
```
Comment (devansh-srv) [inline backend/app/routes/admin_routes.py:63]: what is B_ID?

File: client/src/api/api.js [hunk 2]
```diff
@@ -23,9 +23,11 @@ export const getAdminCredits = () => api.get('/admin/credits');
 export const createAdminCredit = (creditData) => api.post('/admin/credits', creditData);
 export const getBuyerCredits = () => api.get('/buyer/credits');
 export const purchaseCredit = (purchaseData) => api.post('/buyer/purchase', purchaseData);
+export const sellCreditApi = (sellData) => api.patch('/buyer/sell', sellData);
+export const removeSaleCreditApi = (removeData) => api.patch('/buyer/remove-from-sale', removeData);
 export const getPurchasedCredits = () => api.get('/buyer/purchased');
 export const getTransactions = () => api.get('/admin/transactions');
-export const generateCertificate = (purchaseId) => api.get(`/buyer/generate-certificate/${purchaseId}`);
+export const generateCertificate = (creditId) => api.get(`/buyer/generate-certificate/${creditId}`);
```
Comment (devansh-srv) [inline client/src/api/api.js:30]: we should be generating the certificate with  purchaseID only

**Review by devansh-srv — APPROVED** [review]

File: backend/app/routes/admin_routes.py (hunk 1 above)
Comment (R0gue-one) [reply backend/app/routes/admin_routes.py:63]: most probably means Backend_ID (as backend is 1 indexed and contract is 0)

File: client/src/api/api.js (hunk 2 above)
Comment (R0gue-one) [reply client/src/api/api.js:30]: thik hai na yeh toh api call hai, ek hi mtlb hai.
Kuchh dikkat aya tha uss time purchased id se ho nhi rha tha sare chize access shyd.

Comment (devansh-srv) [reply client/src/api/api.js:30]: ok boss

**Pull request merged**